src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -181,6 +181,14 @@ public override void OnEnd(Activity data)
             return;
         }
 
+        // Skip any activities that are not recorded.
+        if (data is { Recorded: false, IsAllDataRequested: false })
```

Comment:
It needs both to be "not recorded"?

Reply:
I did that subconsciously to be honest. I have a vague memory of the OTel docs saying `IsAllDataRequested` should always be false when `Recorded` is false, but for our purposes I think you're right - we can just check `Recorded`. 

Note: The [docs for IsAllDataRequested](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.isalldatarequested?view=net-9.0) say:
> indicates whether this activity should be populated with all the propagation information, as well as all the other properties, such as links, tags, and events